### PR TITLE
Add week 8 contents

### PR DESCRIPTION
**Pull request by AdrianDoM**
This covers: file organisation (sections, namespaces, imports), attributes and simp, defining notation.
It ends with a challenge exercise which consists on learning topology from MIL and then checking your understanding.

### CODE REVIEW

File: Formal2024/Workshops/workshop8.lean [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+postfix:max "⁻¹"  => Inv.inv
+
+-- Note that we can only assign notation to functions that are already defined.
+
+-- The previous lines are equivalent to these, which use the full `notation` command.
+notation:65 lhs:65 " + " rhs:66 => HAdd.hAdd lhs rhs
+notation:50 lhs:51 " = " rhs:51 => Eq lhs rhs
+notation:80 lhs:81 " ^ " rhs:80 => HPow.hPow lhs rhs
+notation:100 "-" arg:100 => Neg.neg arg
+notation:1024 arg:1024 "⁻¹" => Inv.inv arg
```
Comment (SimoneCastellan) [inline Formal2024/Workshops/workshop8.lean:288]: This last bit feels a bit rushed. Maybe we could skip the notation command and just do a couple of examples of new notation, showing the difference of left/right associativity?

**Review by SimoneCastellan — COMMENTED** [review]
Hi, it looks very good! I left just one comment on the notation section, but if you don't have time to change no problem, is no big deal

File: Formal2024/Workshops/workshop8.lean (hunk 1 above)
Comment (AdrianDoM) [reply Formal2024/Workshops/workshop8.lean:288]: Just added some more examples, and explained the precedence number and how it gives left and right associativity.

**Comment by themathqueen** [comment]
This looks good! Exactly what I had in mind!

**Review by themathqueen — APPROVED** [review]

**Pull request merged**